src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -132,6 +135,19 @@ public void run() {
         associations.ifPresent(this::refresh);
     }
 
+
+    /**
+     * Handler to react to network changes.
+     * @param connectionState - A network state
+     */
+    @Override
+    public void accept(NetworkState.ConnectionState connectionState) {
```

Comment:
Let's still cap this at once a day. Maybe track last sync time?

Reply:
I had that same thought but decided not to add it. I think that is fine. I'll change it

Reply:
Done